hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,4 @@
+RELEASE_TYPE: patch
+
+Fixes occasional recursion-limit-exceeded errors at validate- and draw-time for
+deeply nested strategies. Closes: 3671
```

Comment:
As below, I'm pretty sure that this could _only_ happen at validate-time in `@given()`, and we already handled draw-time problems just fine.  (though the bug might have _manifested_ at draw-time, when a concurrent thread edited the recursion depth)

Let's also use the `:issue:` cross-reference syntax to add a hyperlink.